### PR TITLE
dotCMS/core#15509 Upgrade the dot-page-selector.component

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/app/portlets/content-types/form/content-types-form.component.spec.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ import { DotDirectivesModule } from '@shared/dot-directives.module';
 import { DotIconModule } from '@components/_common/dot-icon/dot-icon.module';
 import { DotIconButtonModule } from '@components/_common/dot-icon-button/dot-icon-button.module';
 import { MdInputTextModule } from '@directives/md-inputtext/md-input-text.module';
+import {DotFieldHelperModule} from '@components/dot-field-helper/dot-field-helper.module';
```
Comment (fmontes) [inline src/app/portlets/content-types/form/content-types-form.component.spec.ts]: prettier

File: src/app/portlets/content-types/form/content-types-form.component.ts [hunk 2]
```diff
@@ -145,6 +145,7 @@ export class ContentTypesFormComponent implements OnInit, OnDestroy {
      * @memberof ContentTypesFormComponent
      */
     submitForm(): void {
+        debugger;
```
Comment (fmontes) [inline src/app/portlets/content-types/form/content-types-form.component.ts]: remove

File: src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts [hunk 3]
```diff
@@ -7,16 +7,19 @@ import { DotPageSelectorComponent } from './dot-page-selector.component';
...
 import {
     DotPageSelectorResults,
     DotPageSeletorItem
 } from '@components/_common/dot-page-selector/models/dot-page-selector.models';
 import { LoginService } from 'dotcms-js';
 import { LoginServiceMock } from '../../../../test/login-service.mock';
+import { DotFieldHelperModule } from '@components/dot-field-helper/dot-field-helper.module';
+import { DotMessageService } from '@services/dot-messages-service';
+import { MockDotMessageService } from '@tests/dot-message-service.mock';
+import { Site } from '../../../../../../projects/dotcms-js/src/lib/core/site.service';
```
Comment (fmontes) [inline src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts]: `from 'dotcms-js'`

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
-import { map, pluck, flatMap, toArray, tap, switchMap } from 'rxjs/operators';
+import {map, pluck, flatMap, toArray, switchMap, take} from 'rxjs/operators';
 import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs';
+import {Observable, of} from 'rxjs';
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts]: prettier

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts [hunk 5]
```diff
@@ -240,11 +247,15 @@ export class DotPageSelectorService {
...
 
     private shouldSearchPages(query: string): boolean {
-        const parsedURL = this.parseUrl(query);
-        if ( !parsedURL || this.isReSearchingForHost(query)) {
+        // const parsedURL = this.parseUrl(query);
+        if (!this.isHostAndPath(query) || this.isReSearchingForHost(query)){
             this.currentHost = null;
         }
 
+        // if (!parsedURL || this.isReSearchingForHost(query)) {
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts]: remove

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts [hunk 6]
```diff
@@ -240,11 +247,15 @@ export class DotPageSelectorService {
     }
 
     private shouldSearchPages(query: string): boolean {
-        const parsedURL = this.parseUrl(query);
-        if ( !parsedURL || this.isReSearchingForHost(query)) {
+        // const parsedURL = this.parseUrl(query);
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts]: remove

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts (hunk 5 above)
Comment (fmontes) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts]: remove commented code.

**Pull request merged**